### ISSUE TITLE
False negative: Use of variable in type expression is not reported if its type is a TypeVar or Self

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#9750]
The typing spec says that variables cannot be used within type expressions, but pyright fails to report an error in some cases.

```python
class A:
    @classmethod
    def m1(cls):
        a: cls

    def m2(self):
        a: self
```

By contrast, mypy correctly flags these errors.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.393.
